.circleci/config.yml

```diff
@@ -114,13 +137,52 @@ workflows:
             - git-user-ops
             - github-bot-public
 
-  on-main:
+  on-main-branch:
     # On all pushes to main, except when it's a scheduled pipeline.
     when:
       and:
         - equal: [ main, << pipeline.git.branch >> ]
         - not:
             equal: [ scheduled_pipeline, << pipeline.trigger_source >> ]
     jobs:
-      - deploy-snapshot:
+      - publish:
+          snapshot: true
+          context: maven-central-publishing
+
+  release-train:
+    when:
+      or:
+        - and:
+            - equal: [ scheduled_pipeline, << pipeline.trigger_source >> ]
+            - equal: [ "release-train", << pipeline.schedule.name >> ]
+        # The release-train can be manually triggered by passing the 
+        # "action" parameter with value "bump".
+        - equal: [ bump, << pipeline.parameters.action >> ]
+    jobs:
+      - revenuecat/automatic-bump
+
+  on-release-branch:
+    when:
+      and:
+        - matches: { pattern: "^release/.+$", value: << pipeline.git.branch >> }
```

Comment:
Hmm interesting... We've usually had `aliases` for these...I guess this applies to the whole workflow, which might be nicer than having to copy the alias on each job, so maybe this is ok.

I have wondered before if it would be possible to move those aliases to our CircleCI orb but haven't taken a look. Not sure if it would be possible with this... In any case, that's something we can do later.

Reply:
Yea I was wondering what you'd think of this 🙂 I personally think this is more readable than having `when` clauses (through aliases) on each job. It's easier to follow the logic this way. In any case I added some "header docs" to each workflow. 

Looking quickly at the CircleCI docs, I don't think it's possible for an orb to provide aliases. GitLab CI/CD has the concept of [includes](https://docs.gitlab.com/ee/ci/yaml/includes.html), which would allow us to define aliases in a separate YAML file in a central git repo, and then include that in all our other repos. It [doesn't seem like](https://support.circleci.com/hc/en-us/articles/360056463852-Can-I-split-a-config-into-multiple-files) CircleCI has anything like that though.